libcontainer/cgroups/fs/devices.go

```diff
@@ -26,7 +26,11 @@ func (s *DevicesGroup) Apply(d *cgroupData) error {
 }
 
 func (s *DevicesGroup) Set(path string, cgroup *configs.Cgroup) error {
-	if system.RunningInUserNS() {
+	pids, err := cgroups.GetPids(path)
+	if err != nil {
+		return err
+	}
+	if system.RunningInUserNS(pids[0]) {
```

Comment:
Ah dammit. `cgroupData` has the PID of the process being applied. Maybe you could change the calling convention in some way to include the PID? I don't like using `cgroups.GetPids` for this -- what happens if someone is sharing the cgroup?